Blog post about simplified scripts in 2.0.0

mikewalch opened the pull request:
* Also minor change to post template

ctubbsii commented:
Haven't looked at the content yet, but we should wait until 2.0.0 is released before this blog is posted. Things could still change.

mikewalch commented:
@ctubbsii, I thought we could just put it out there. If things change, they'll probably be small but I will update the blog post.

ctubbsii commented:
In the ASF, there is a clear line drawn between released software and unreleased software, and promoting or marketing unreleased software in the ASF is a big no-no. Even beta or alpha versions must be voted on and released.

Development should happen in the developer forums (mailing list). The blogs on the website are not developer forums. It's intended for users.

I think it's fine to post-date the blog into the far future, so it doesn't get published by Jekyll, or make it into the RSS feed for blogs, and then update the date after release. Or, just hold the PR open until after the release. This is a good staging area, and it'd be good to have some blogs ready to go to market 2.0.0, after we approve a release.

ctubbsii commented:
For reference, [this page][1] discusses/forbids distributing material or advertising which encourages the use of unreleased software outside the development community.

[1]: https://www.apache.org/dev/release-distribution.html#unreleased

joshelser commented:
Aside from the last paragraph, I really don't think we have to wait to post this entry.

Presenting work that has been done in Accumulo is fine. It shows activity in the project and defeats the nay-sayer's that claim the project is dead. That can be done in a way that doesn't encourage users to not use official releases.

Skimming the content on the scripts themselves, it looks like a really nice post.

ctubbsii commented:
If the last paragraph is dropped or changed so that it doesn't encourage users to use unreleased software, and the title is changed to "Simpler scripts and configuration coming in 2.0.0", to clearly document it as an anticipated future feature in the RSS feed, then it should be fine. I still think it's prudent to wait (gives us more to market the 2.0.0 release with later, and I don't think there's a rush on this). However, those changes would sufficiently address my concern.

mikewalch commented:
I pushed another commit addressing concerns. The release notes will market all changes when release is made.  They could even link to this blog post when discussing this change. I will post it today unless I hear any more concerns.

joshelser commented on _posts/blog/2016-11-16-simpler-scripts-and-config.md:
```diff
@@ -0,0 +1,163 @@
+Core Commands:
+  create-config                  Create Accumulo configuration
+  build-native                   Build Accumulo native libraries
+  init                           Initializes Accumulo
```
I'm just noticing the inconsistency in the description:

> Create Accumulo configuration

> Initializes Accumulo

I assume this would be a tweaking of the scripts themselves and then later updating this post?

Personally, because these are all actions, they should all be 3rd-person:

> Creates Accumulo configuration
> Builds Accumulo native libraries
> Prints Accumulo version

ctubbsii replied:
Descriptions [should use][1] third-person, active voice, indicative mood, while instructions should use second-person, active voice, imperative mood.

[1]: http://www.technicalauthoring.com/wiki/index.php/Using_the_correct_language

joshelser commented:
> I will post it today unless I hear any more concerns.

:+1:

Just the one minor grammatical request on the action descriptions from me. I think this looks good.

ctubbsii commented on _posts/blog/2016-11-16-simpler-scripts-and-config.md:
```diff
@@ -0,0 +1,163 @@
+---
+
+For the upcoming 2.0.0 release, Accumulo's scripts and configuration were refactored (in 
+[ACCUMULO-4490]) to make Accumulo easier to use. While Accumulo's documentation (i.e. the user
```
Better to remove the parenthetical and change it to `[were refactored][ACCUMULO-4490]`.

ctubbsii commented on _posts/blog/2016-11-16-simpler-scripts-and-config.md:
```diff
@@ -0,0 +1,163 @@
+
+### Fewer scripts
+
+Before 2.0.0, the `bin/` directory of Accumulo's distribution contained about 20 scripts:
```
s/distribution/binary tarball/

mikewalch replied:
Fixed

ctubbsii commented on _posts/blog/2016-11-16-simpler-scripts-and-config.md:
````diff
@@ -0,0 +1,163 @@
+For 2.0.0, all 'accumulo' commands were given a short description and organized into the groups.
+
+```
+$ ./accumulo-2.0.0/bin/accumulo help
````
Should you explain that the usage will be limited if you don't first set up the classpath with create-config?

mikewalch replied:
Good idea.  Fixed

ctubbsii commented on _posts/blog/2016-11-16-simpler-scripts-and-config.md:
````diff
@@ -0,0 +1,163 @@
+```
+
+[ACCUMULO-4490]: https://issues.apache.org/jira/browse/ACCUMULO-4490
+[INSTALL.md]: https://github.com/apache/accumulo/blob/master/INSTALL.md
````
Note: Link will need to be updated upon release, to point to the tag or commit, so it doesn't get stale.

mikewalch replied:
Ok.. will update after release.
